python/rmm/tests/test_rmm.py

```diff
@@ -691,3 +691,27 @@ def callback(nbytes: int) -> bool:
     with pytest.raises(MemoryError):
         rmm.DeviceBuffer(size=int(1e11))
     assert retried[0]
+
+
+def test_dev_buf_circle_ref_dealloc():
```

Comment:
Can we also add a test that uses an `UpstreamMemoryResource`, since that case was found to be problematic?

Reply:
What initially allowed me to see the bug, is actually an existing well named test : `test_mr_devicebuffer_lifetime`. It uses a `PoolMemoryResource` that inherits from the `UpstreamResourceAdaptor`. This perfectly demonstrate/test the issue. Should we add an other test?

Reply:
Yes - although let's document the intent to use a resource with an upstream (not specifically a pool memory resource)